src/reducers/comments.spec.tsx

```diff
@@ -19,35 +25,80 @@ describe(__filename, () => {
 
   describe('beginComment', () => {
     it('begins a comment by key', () => {
-      const fileName = 'manifest.json';
-      const line = 123;
-      const versionId = 1;
+      const params = { versionId: 1, fileName: 'manifest.json', line: 123 };
 
-      const state = reducer(
-        undefined,
-        actions.beginComment({ versionId, fileName, line }),
+      const state = reducer(undefined, actions.beginComment(params));
+
+      expect(state.byKey[createCommentKey(params)]).toEqual(
+        createCommentInfo({ beginNewComment: true }),
       );
+    });
 
-      expect(
-        state.byKey[createCommentKey({ versionId, fileName, line })],
-      ).toEqual(createCommentInfo({ beginNewComment: true }));
+    it('aborts saving a comment by key', () => {
+      const params = { versionId: 1, fileName: 'manifest.json', line: 123 };
+
+      let state;
+
+      // Imagine that this happened some time in the past.
+      state = reducer(state, actions.beginSaveComment(params));
+      // Imagine that the user opened a comment form again.
+      state = reducer(state, actions.beginComment(params));
+
+      expect(state.byKey[createCommentKey(params)]).toMatchObject({
+        savingComment: false,
+      });
+    });
+  });
+
+  describe('beginSaveComment', () => {
+    it('begins saving a comment by key', () => {
+      const params = { versionId: 1, fileName: 'manifest.json', line: 123 };
+
+      const state = reducer(undefined, actions.beginSaveComment(params));
+
+      expect(state.byKey[createCommentKey(params)]).toEqual(
+        createCommentInfo({ savingComment: true }),
+      );
+    });
+  });
+
+  describe('abortSaveComment', () => {
+    it('aborts saving a comment by key', () => {
+      const params = { versionId: 1, fileName: 'manifest.json', line: 123 };
+
+      let state;
+
+      state = reducer(state, actions.beginSaveComment(params));
+      state = reducer(state, actions.abortSaveComment(params));
+
+      expect(state.byKey[createCommentKey(params)]).toEqual(
+        createCommentInfo({ savingComment: false }),
+      );
     });
   });
 
   describe('finishComment', () => {
     it('finishes a comment by key', () => {
-      const fileName = 'manifest.json';
-      const line = 123;
-      const versionId = 1;
+      const params = { versionId: 1, fileName: 'manifest.json', line: 123 };
 
-      const state = reducer(
-        undefined,
-        actions.finishComment({ fileName, line, versionId }),
+      const state = reducer(undefined, actions.finishComment(params));
+
+      expect(state.byKey[createCommentKey(params)]).toEqual(
+        createCommentInfo({ beginNewComment: false }),
       );
+    });
 
-      expect(
-        state.byKey[createCommentKey({ versionId, fileName, line })],
-      ).toEqual(createCommentInfo({ beginNewComment: false }));
+    it('aborts saving a comment', () => {
```

Comment:
Should this say "finishes saving a comment"? We're not actually aborting the save here.